src/core_init_atmosphere/mpas_kd_tree.F

```diff
@@ -0,0 +1,394 @@
+module mpas_kd_tree
+
+   !***********************************************************************
+   !
+   !  module mpas_kd_tree
+   !
+   !> \brief   MPAS KD-Tree module
+   !> \author  Miles A. Curry
+   !> \date    01/28/20
+   !> A KD-Tree implementation to create and search perfectly balanced
+   !> KD-Trees.
+   !>
+   !> Use `mpas_kd_type` dervied type to construct points for mpas_kd_construct:
+   !>
+   !> real, dimension(:,:), allocatable :: array
```

Comment:
For consistency, we might consider using `real (kind=RKIND)` in this example rather than simply `real`. Also, in the example below, after the elements of `array` are copied to the `point` members of the `points` array, we can `deallocate(array)` to make clear to readers that the memory used by `array` should be freed to prevent a memory leak.